sw/host/opentitanlib/src/image/image.rs

```diff
@@ -133,6 +136,77 @@ impl Image {
         Ok(())
     }
 
+    /// Adds an extension to the signed region of this `Image`.
+    pub fn add_signed_manifest_extensions(&mut self, spec: ManifestExtSpec) -> Result<()> {
+        self.add_manifest_extensions(&spec.signed_region, spec.source_path())
+    }
+
+    /// Adds an extension to the unsigned region of this `Image`.
+    pub fn add_unsigned_manifest_extensions(&mut self, spec: ManifestExtSpec) -> Result<()> {
+        self.add_manifest_extensions(&spec.unsigned_region, spec.source_path())
+    }
+
+    fn add_manifest_extensions(
+        &mut self,
+        entries: &[ManifestExtEntrySpec],
+        relative_path: Option<&Path>,
+    ) -> Result<()> {
+        let manifest = self.borrow_manifest()?;
+
+        // A copy of the extension table since we can't borrow as mutable.
+        let mut ext_table = manifest.extensions.entries;
+
+        for spec in entries {
+            let entry = ManifestExtEntry::from_spec(spec, relative_path)?;
+            let entry_id = entry.header().identifier;
+
+            // Update the offest in the extension table.
+            let ext_table_entry = ext_table
+                .iter_mut()
+                .find(|e| e.identifier == entry_id)
+                .ok_or(ImageError::NoExtensionTableEntry(entry_id))?;
+
+            // If the extension already exists, overwrite it, else append it to the end of the
+            // image.
+            let offset = if ext_table_entry.offset != 0 {
+                ext_table_entry.offset as usize
+            } else {
+                // Align the extension offset to the next word boundary.
+                let misalignment = self.size % align_of::<u32>();
+                if misalignment != 0 {
+                    log::warn!(
+                        "End of image is not word aligned, forcing alignment for extension 0x{:x}",
+                        entry_id
+                    );
+                    self.size
+                        .checked_add(align_of::<u32>() - misalignment)
+                        .ok_or(ImageError::ExtensionOverflow)?
+                } else {
+                    self.size
+                }
+            };
+            ext_table_entry.offset = offset as u32;
```

Comment:
This approach LGTM. IIRC I used `LayoutVerified` so that I can convert the SPHINCS+ key and signature to their respective structs and pass them from `opentitantool/command/image.rs` to a method in `opentitanlib/image`.